README.md

```diff
@@ -1,68 +1,212 @@
 
-## Setup
+#### Table of Contents
 
-To install dependencies:
+- [Setup](#setup)
+- [Getting Started](#getting-started)
+- [Development Mode/Local dev](#development-mode)
+- [Environment Variables](#environment-variables)
+- [Testing](#testing)
 
-```bash
+## Setup
+veritable-ui depends on a few external services:
+- [veritable-cloudagent](https://github.com/digicatapult/veritable-cloudagent) - APIs for managing connections, credentials and messages.
+- `keycloak` - runs as a docker image for each node that handles authentication and users.
+- `docker` - for orchestrating multiple nodes and running other dependencies such as database (PostgreSQL) along with node specific environment variables
+
+### Prerequisites
+> :warning: last updated: 01/08/2024
+- docker v27.0.3+
+- npm 10.0.0+
+- node 22.0.0+
+- keycloak:25.0.2+ - included in the docker-compose.yml
+- veritable-cloudagent@latest - last used -> v0.9.42
+
+## Getting started
+Ensure you're running the correct version of npm, then install dependencies using:
+```sh
 npm install
 ```
 
-And then to building the `tsoa` routes:
-
-```bash
+After all packages have been installed run the below command which will create two files under `/src` directory: `src/routes.ts`, `src/swagger.json`
+```sh
 npm run tsoa:build
 ```
 
-To setup service dependencies make sure there's at a minimum an empty `.env` along with `COMPANY_PROFILE_API_KEY` key. It can be any string ->`COMPANY_PROFILE_API_KEY=a`. `.env` file must be in the root directory of this repository and then run:
+Once TSOA build has completed, then run a typescript compiler
+```sh
+npm run build
+```
 
-```bash
-docker compose up -d
+> :memo: Note that this is just the service on itself without other dependencies like keycloak or database, for local development please refer to the section below
+
+## Development mode
+Before reading this please make sure that you have executed all the tasks from **Getting Started** section (installation and build), should have a directory `/build` and make sure there is at a minimum an empty `.env` at the root of project containing the below:
+```sh
+COMPANY_PROFILE_API_KEY=apikey
 ```
 
-Followed by database migrations
+After creating `.env` file you can execute `docker` which will bring all the nodes and required services for local development
+```sh
+docker compose up -d
+```
+> :memo: If doing this for the first it might take some time to pull all the images
 
-```bash
+Followed by the database migrations
+```sh
 npm run db:migrate
 ```
 
-Finally to run:
+> :memo: Optional: you can seed some mock data for rendering some items in tables and etc. run: `npm run db:seed`
 
-```bash
-npm run dev
+Also, we need to create some credential definitions along with `veritable-cloudagent` instance
+```sh
+npm run dev:init
```

Comment:
you don't technically need to do this, but I don't mind leaving it in if you'd prefer